comparison of integer expressions of different signedness: ‘int’

# Issue

dash/src/canbus/elm327.cpp:144:19: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::__cxx11::basic_string<char>::size_type’ {aka ‘unsigned int’} [-Wsign-compare]
     for(int i=0; i<resp_str.length()/2; i++){

--- a/src/canbus/elm327.cpp
+++ b/src/canbus/elm327.cpp
@@ -141,7 +141,7 @@ QCanBusFrame elm327::receive()
 
     retFrame->setFrameId(0x7E8);
     QByteArray payload = QByteArray::fromHex("0000000000000000");
-    for(int i=0; i<resp_str.length()/2; i++){
+    for( std::vector<int>::size_type i=0; i < resp_str.length()/2; i++){
         payload[i+1] = std::stoi(resp_str.substr(2*i, 2), nullptr, 16);
     }
     payload[0] = resp_str.length()/2;

